ext/nnstreamer/tensor_filter/tensor_filter_python_core.cc

```diff
@@ -0,0 +1,718 @@
+/**
+ * Copyright (C) 2019 Samsung Electronics Co., Ltd. All rights reserved.
+ * Copyright (C) 2019 Dongju Chae <dongju.chae@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file   tensor_filter_python_core.cc
+ * @author Dongju Chae <dongju.chae@samsung.com>
+ * @date   03/29/2019
+ * @brief  connection with python libraries.
+ * @bug     No known bugs.
+ */
+
+#include "tensor_filter_python_core.h"
+#include <nnstreamer_plugin_api.h>
+
+/**
+ * @brief Macro for debug mode.
+ */
+#ifndef DBG
+#define DBG FALSE
+#endif
+
+std::map <void*, PyArrayObject*> PYCore::outputArrayMap;
+
+/**
+ * @brief	PYCore creator
+ * @param	_script_path	: the logical path to '{script_name}.py' file
+ * @note	the script of _script_path will be loaded simultaneously
+ * @return	Nothing
+ */
+PYCore::PYCore (const char* _script_path, const char* _custom)
+  : script_path(_script_path), module_args(_custom)
+{
+  /**
+   * To fix import error of python extension modules
+   * multiarray.x86_64-linux-gnu.so: undefined symbol: PyExc_SystemError)
+   */
+#if PY_VERSION_HEX >= 0x03000000
+  dlopen("libpython3.5m.so.1", RTLD_LAZY | RTLD_GLOBAL);
+  dlopen("libpython3.6m.so.1", RTLD_LAZY | RTLD_GLOBAL);
```

Comment:
Q1. Do you really need to load both libraries? What if both are available? Would it behave as you've wanted?

Reply:
The reason why I added two libraries is that some test platforms have different python3 versions.
(e.g., Tizen: 3.6, Ubuntu: 3.5)
I will handle this to avoid dlopen()-ing both libraries using a return value.